File: user-service/apps/user-api/src/main/java/com/yellowsunn/userservice/config/TokenConfiguration.java
package com.yellowsunn.userservice.config;

import com.yellowsunn.common.utils.token.AccessTokenHandler;
import com.yellowsunn.common.utils.token.RefreshTokenHandler;
import java.time.Duration;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class TokenConfiguration {

    @Bean
    public AccessTokenHandler accessTokenHandler(
            @Value("${token.access.secret}") String secret,
            @Value("${token.access.expiration}") Duration expiration
    ) {
        return new AccessTokenHandler(secret, expiration);
    }

    @Bean
    public RefreshTokenHandler refreshTokenHandler(
            @Value("${token.refresh.secret}") String secret,
            @Value("${token.refresh.expiration}") Duration expiration
    ) {
        return new RefreshTokenHandler(secret, expiration);
    }
}

File: apigateway-service/src/main/kotlin/com/yellowsunn/apigatewayservice/config/TokenConfiguration.kt
package com.yellowsunn.apigatewayservice.config

import com.yellowsunn.common.utils.token.AccessTokenParser
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class TokenConfiguration {
    @Bean
    fun accessTokenParser(
        @Value("\${token.access.secret}") secret: String,
    ): AccessTokenParser {
        return AccessTokenParser(secret)
    }
}
